### PR TITLE
[CALCITE-6392] Support all PostgreSQL 14 date/time patterns for to_da…

### PR DESCRIPTION
**Pull request by normanj-bitquill**
…te/to_timestamp

* First phase, mostly reorganizing classes
* Does not yet implement to_date or to_timestamp
* Fixed up fill mode handling
* Isolated the unit test class so that it can make Locale changes safely
* Introduced some new classes split out handling of some formats strings

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/util/format/postgresql/DateStringFormatPattern.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    @Override public DayOfWeek getValueFromDateTime(ZonedDateTime dateTime) {
+      return dateTime.getDayOfWeek();
+    }
+
+    @Override public String getDisplayName(DayOfWeek value, TextStyle textStyle,
+        boolean haveFillMode, Locale locale) {
+      final String formattedValue = value.getDisplayName(textStyle, locale);
+
+      if (!haveFillMode && textStyle == TextStyle.FULL) {
+        return String.format(locale, "%-9s", formattedValue);
```
Comment (mihaibudiu) [inline core/src/main/java/org/apache/calcite/util/format/postgresql/DateStringFormatPattern.java:61]: why 9 exactly?

**Review by mihaibudiu — COMMENTED** [review]
This looks in large part like a refactoring of code that I have reviewed recently.

File: core/src/main/java/org/apache/calcite/util/format/postgresql/DateStringFormatPattern.java (hunk 1 above)
Comment (normanj-bitquill) [reply core/src/main/java/org/apache/calcite/util/format/postgresql/DateStringFormatPattern.java:61]: From the PostgreSQL documentation, day of week names are blank padded to 9 characters.
https://www.postgresql.org/docs/14/functions-formatting.html#FUNCTIONS-FORMATTING-DATETIME-TABLE

Look for `DAY`, `Day` or `day`.

Comment (mihaibudiu) [reply core/src/main/java/org/apache/calcite/util/format/postgresql/DateStringFormatPattern.java:61]: I would put that into a comment.

Comment (normanj-bitquill) [reply core/src/main/java/org/apache/calcite/util/format/postgresql/DateStringFormatPattern.java:63]: @mihaibudiu Added comments to explain this in the code.

**Review by mihaibudiu — APPROVED** [review]

**Comment by mihaibudiu** [comment]
Merging will require the commits to be squashed

**Comment by normanj-bitquill** [comment]
@mihaibudiu The commits have been merged.

**Pull request merged**